msteams-platform/resources/schema/manifest-schema-dev-preview.md

```diff
@@ -1001,6 +1002,23 @@ The `extensions.runtimes` array configures the sets of runtimes and actions that
 
 To use `extensions.runtimes`, see [create add-in commands](/office/dev/add-ins/develop/create-addin-commands-unified-manifest), [configure the runtime for a task pane](/office/dev/add-ins/develop/create-addin-commands-unified-manifest#configure-the-runtime-for-the-task-pane-command), and [configure the runtime for the function command](/office/dev/add-ins/develop/create-addin-commands-unified-manifest#configure-the-runtime-for-the-function-command).
 
+### extensions.contentRuntimes
+
+The `extensions.contentRuntimes` array configures pages of content that are embedded in an Excel or PowerPoint document. 
+
+|Name| Type| Maximum size | Required | Description|
+|---|---|---|---|---|
+|`id`| String | 64 characters | ✔️ | Specifies the ID for runtime. Must be unique across all `runtimes` and `contentRuntimes` objects. |
+|`code`| Object | | ✔️ | Specifies the location of content for the runtime. |
+|`code.page`| URL | | ✔️ | The full URL of the page that contains the content that is embedded in the document. |
+|`requestedWidth`| Number | | | The desired width between 32 and 1000 pixels. If this property isn't used, the Office application determines the width.|
+|`requestedHeight`| Number | | | The desired height between 32 and 1000 pixels. If this property isn't used, the Office application determines the height.|
+|`disableSnapshot` | boolean | | | Specifies whether a snapshot image of your content add-in is saved with the host document. Default value is `false`, so a snapshot is saved. Set to `true` to prevent a snapshot from being saved.|
+|`requirements`| Object | | | Specifies the scopes, formFactors, and Office JavaScript Library requirement sets that must be supported on the Office client in order for the content to be embedded in the document. For more information, see [Specify Office Add-in requirements in the unified manifest for Microsoft 365](/office/dev/add-ins/develop/requirements-property-unified-manifest).|
+|`requirements.capabilities`| Array | | | Don't use when the grandparent is a content runtime object. |
```

Comment:
It cannot be removed because the "contentRuntimes.requirements" property uses the standard data type of "requirements" properties, so "capabilities" will appear in Intellisense even when the grandparent is a content runtimes object.

Reply:
> Please check whether this property can be removed since we are adding it under contentRuntimes.

It cannot be removed because the "contentRuntimes.requirements" property uses the standard data type of "requirements" properties, so "capabilities" will appear in Intellisense even when the grandparent is a content runtimes object.

Reply:
Even if it appears in intellisense, the developer can decide to add or ignore it. Instead of adding it in the property table, can we add a note that 'requirements.capabilities and requirements.formFactors are not required for contentRuntimes?

Reply:
> Even if it appears in intellisense, the developer can decide to add or ignore it. Instead of adding it in the property table, can we add a note that 'requirements.capabilities and requirements.formFactors are not required for contentRuntimes?

We need to say that they should not be used, not that they are "not required". They aren't **required** in ANY scenario. They are always optional. But in most grandparent properties, they will have an effect if they are used. In the case of contentRuntimes, they won't.

Reply:
Can we mention the same as note then? My concern of adding it in properties table is, some users will add and check properties in the list intuitively before reading the description. Having the properties not to be added in the table may cause confusion. What are your thoughts?

Reply:
> Can we mention the same as note then? My concern of adding it in properties table is, some users will add and check properties in the list intuitively before reading the description. Having the properties not to be added in the table may cause confusion. What are your thoughts?

Please clarify what you mean by "same as note". 

I think users that just check properties in the list will note that "capabilities" and "formFactors" are not checked in the **Required** column. At any rate, we should not design our documentation to accommodate people who don't read the documentation. It is reasonable to assume that people who are looking at these rows of the table are going to read the description. Also, note that people don't use the "requirements" property based only on what they see in this brief reference documentation. They will be educated in the use of that property by reading the conceptual documentation, especially [Specify Office Add-in requirements in the unified manifest for Microsoft 365](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/requirements-property-unified-manifest) which will have a section about "extensions.contentRuntimes.requirements".

Reply:
Thanks for adding clarity, @Rick-Kirkham. Give me some time to analyze the user experience of having the properties in the table and having them as a note at the end of the table. Will revert soon.

Reply:
Hi @Rick-Kirkham,
As suggested by you, we can add it in the table. Can we edit the description alone since we have never used the word 'grandparent' in our documents?
**Suggestion:** Don't add this property for 'contentRuntimes' object.

Reply:
> Hi @Rick-Kirkham, As suggested by you, we can add it in the table. Can we edit the description alone since we have never used the word 'grandparent' in our documents? **Suggestion:** Don't add this property for `contentRuntimes` object. (or) This property (though present under `requirements` for other objects) must not be added for `contentRuntimes` object.

Yes. I'll use your second suggestion. Thanks!